src/Microsoft.AspNet.Mvc.Core/Formatters/FormatterCollection.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Mvc.Formatters
+{
+    public class FormatterCollection<TFormatter> : List<TFormatter>
```

Comment:
And if you really still want to be able to call `RemoveAll` you can use this constructor: https://msdn.microsoft.com/en-us/library/ms132401(v=vs.110).aspx and keep a reference to the list you create.